src/AcceptanceTests/When_session_is_accessed_in_hosted_service_start.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace AcceptanceTests
+{
+    using Microsoft.Extensions.Hosting;
+    using NUnit.Framework;
+    using NServiceBus;
+    using Microsoft.Extensions.DependencyInjection;
+    using System.Threading.Tasks;
+    using System.Threading;
+    using System;
+
+    [TestFixture]
+    public class When_session_is_accessed_in_hosted_service_start
+    {
+        [Test]
+        public async Task Should_be_available_when_configured_after_NServiceBus()
+        {
+            var host = Host.CreateDefaultBuilder()
+                .UseNServiceBus(hostBuilderContext =>
+                {
+                    var endpointConfiguration = new EndpointConfiguration("MyEndpoint");
+                    endpointConfiguration.SendOnly();
+                    endpointConfiguration.UseTransport<LearningTransport>();
+                    return endpointConfiguration;
+                })
+                .ConfigureServices((ctx, serviceProvider) => serviceProvider.AddHostedService<HostedServiceThatAccessSessionInStart>())
+                .Build();
+
+            await host.StartAsync();
+            await host.StopAsync();
```

Comment:
The clean stopping of the service would have to happen in the TearDown stage.

Reply:
There are multiple tests in here so not sure how we can use the TearDown?

Reply:
The best way is try finally like I did in the other PR. Teardown doesn't work because you can only shut down the built host which makes setup and teardown painful

Reply:
When doing this a test failure causes a weird

```    System.AggregateException : One or more hosted services failed to stop.
      ----> System.NullReferenceException : Object reference not set to an instance of an object.
```

from the stop method so I'm not sure this adds any values since it prevents the test to show the real failure make it hard to diagnose?

Seems like the generic host doesn't support calling stop if start throws. I will keep the existing code that works and do a separate investigation